Plutarch/Builtin.hs

```diff
@@ -240,6 +253,12 @@ instance PIsData a => PIsData (PBuiltinList (PAsData a)) where
   pfromData x = punsafeCoerce $ pasList # pforgetData x
   pdata x = punsafeBuiltin PLC.ListData # x
 
+newtype Helper2 f a s = Helper2 (PAsData (f a) s)
+
+instance PIsData (PBuiltinList PData) where
```

Comment:
what's up with the complexity here?

Reply:
This is how you do type-level functions in Haskell lol. With dependent types this would have been an order of magnitude less complex, unfortunately we do not have dependent types. I still think this is worth it because the transformation describes here is safe and still zero-cost.

Reply:
well yeah, but you could've also done this with the builtins yeah? I assume this is safer? In which case, all good I suppose.

Reply:
I don't think there's any other way of doing this that is as efficient that isn't unsafe.